account_invoice_import/wizard/account_invoice_import.py

```diff
@@ -205,26 +205,43 @@ def _prepare_create_invoice_vals(self, parsed_inv):
                 static_vals = {}
             for line in parsed_inv['lines']:
                 il_vals = static_vals.copy()
+                if line.get('name'):
+                    il_vals['name'] = line['name']
+                elif line.get('description'):
+                    il_vals['name'] = line['description']
+                elif not il_vals.get('name'):
+                    il_vals['name'] = _('MISSING DESCRIPTION')
                 if config.invoice_line_method == 'nline_auto_product':
-                    product = bdio._match_product(
+                    # Check if we have an invoice2data or xml file
+                    if line.get('description'):
+                        line['product'] = {
+                            'description': line.get('description')
+                        }
+                    match = bdio._match_product(
                         line['product'], parsed_inv['chatter_msg'],
-                        seller=partner)
-                    fposition_id = partner.property_account_position.id
-                    il_vals.update(
-                        ailo.product_id_change(
-                            product.id, product.uom_id.id, type='in_invoice',
-                            partner_id=partner.id,
-                            fposition_id=fposition_id,
-                            company_id=company.id)['value'])
-                    il_vals['product_id'] = product.id
+                        seller=partner, return_dict=True)
+                    if match and match.get('product'):
```

Comment:
You seem to introduce the option that your template parser fails to find the product. I don't see where in the ticket this is required, and I don't think this is a desirable change in the mechanism. Can you raise this as a template failure instead?

Reply:
The ticket described that it should be possible to match products, it does not say it is required to map products, so that's why i interpreted it this way. Let me ask by Els what we should do.

Reply:
Asked Els, we need to add the line without an product if there is no match found.